kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -42,11 +42,9 @@ def __init__(self, name, init_config, agentConfig, instances=None):
         super(KubernetesState, self).__init__(name, init_config, agentConfig, instances=generic_instances)
 
         self.send_pod_phase_service_checks = is_affirmative(instance.get('send_pod_phase_service_checks', True))
-        self._deprecations['send_pod_phase_service_checks'] = [
-            False,
-            "DEPRECATION NOTICE: pod phase service checks are deprecated. Please set "
-            "`send_pod_phase_service_checks` to false and rely on corresponding gauges instead",
-        ]
+        if self.send_pod_phase_service_checks:
+            self.warning("DEPRECATION NOTICE: pod phase service checks are deprecated. Please set "
```

Comment:
Calling instance methods like `warning` in the `__init__` could be risky if the object weren't properly initialized... However at this stage, after the `super()`, this doesn't look too harmful to me. Also, the constructor is unlikely to fail after the warning.